server/bleep/src/webserver/answer/prompts.rs

```diff
@@ -1,102 +1,86 @@
-        "schema": {
-          "name": "proc",
-          "args": [
-            {
-              "name": "QUERY",
-              "type": "STRING",
-              "examples": ["find all the functional react components",
-                            "where are error types"],
-            },
-            {
-              "name": "PATH ALIAS FOR EACH FILE",
-              "type": "INT[]",
-              "examples": [[2, 3, 7, 10]],
-            },
-          ],
-        },
+#[derive(Debug)]
+struct Tool {
+    name: &'static str,
+    description: &'static str,
+    schema: &'static str,
+    examples: &'static str,
+}
+
+impl fmt::Display for Tool {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        writeln!(f, "{}:", self.name)?;
+        writeln!(f, "\tdescription: {}", self.description)?;
+        writeln!(f, "\tschema: {}", self.schema)?;
+        writeln!(f, "\texamples: {:?}", self.examples)?;
+        Ok(())
+    }
+}
+
+pub fn system(paths: &Vec<String>) -> String {
+    let tools = [
+      Tool {
+        name: "codeSearch", 
+        description: "Search the contents of files in a codebase semantically. Results will not necessarily match search terms exactly, but should be related.", 
+        schema: "{\"name\": \"code\", \"args\": [SEARCH_TERMS // str]}", 
+        examples: "[[\"code\", \"backend error types\"], [\"code\", \"react functional components\"]]"
+          },
+      Tool {
+        name: "pathSearch", 
+        description: "Search the pathnames in a codebase. Results may not be exact matches, but will be similar by some edit-distance. Use when you want to find a specific file or directory.", 
+        schema: "{\"name\": \"path\", \"args\": [SEARCH_TERMS // str]}", 
+        examples: "[[\"path\", \"server/src\"], [\"path\", \".tsx\"], [\"path\", \"examples/android\"]]"
       },
-      {
-        "title": "No tool left to take",
-        "description": "Answer the user's query. This is the final step in the process. You should use the information gathered from the other tools to answer the query.",
-        "schema": {
-          "name": "none",
-          "args": [
-              {
-                "name": "RELEVANT PATH ALIASES",
-                "type": "INT[]",
-                "examples": [
-                  [1], [2, 5], [3]
-                ],
-                "description": "The path alias of the files you want to cite.",
-              }
-          ],
-        },
-      }
-    ]);
+      Tool {
+        name: "processFiles", 
+        description: "Read one or more files and extract the line ranges which are relevant to the search terms.", 
+        schema: "{\"name\": \"proc\", \"args\": [SEARCH_TERMS // str, ARRAY_OF_PATH_ALIASES // int[]}", 
+        examples: "[[\"proc\", \"find all the functional react components\", [2, 5]], [\"proc\", \"where are error types\", [0]], [\"proc\", \"gitoxide initialisation\", [2, 5, 8]]]"
```

Comment:
Do we want the `Tool::name` to match the `name` given in `Tool::schema`? In this case here, we see `processFiles`, but in the schema and examples, the tool is named `proc`.

Reply:
Not necessarily. We want the `name` field to be descriptive, and the `name` in `Tool::schema` to be short.